xray/commands/curation/audit.go

```diff
@@ -0,0 +1,504 @@
+	Action            string   `json:"action"`
+	ParentName        string   `json:"direct_dependency_package_name"`
+	ParentVersion     string   `json:"direct_dependency_package_version"`
+	BlockedPackageUrl string   `json:"blocked_package_url"`
+	PackageName       string   `json:"blocked_package_name"`
+	PackageVersion    string   `json:"blocked_package_version"`
+	BlockingReason    string   `json:"blocking_reason"`
+	DepRelation       string   `json:"dependency_relation"`
+	PkgType           string   `json:"type"`
+	Policy            []Policy `json:"policies"`
+}
+
+type Policy struct {
+	Policy    string `json:"policy"`
+	Condition string `json:"condition"`
+}
+
+type PackageStatusTable struct {
+	Status            string        `col-name:"Action"`
+	ParentName        string        `col-name:"Direct Dependency\nPackage Name"`
+	ParentVersion     string        `col-name:"Direct Dependency\nPackage Version"`
+	BlockedPackageUrl string        `col-name:"Blocked Package URL"`
+	PackageName       string        `col-name:"Blocked Package\nName"`
+	PackageVersion    string        `col-name:"Blocked Package\nVersion"`
+	BlockingReason    string        `col-name:"Blocking Reason"`
+	PkgType           string        `col-name:"Package Type"`
+	Policy            []policyTable `embed-table:"true"`
+}
+
+type policyTable struct {
+	Policy    string `col-name:"Violated Policy\nName"`
+	Condition string `col-name:"Violated Condition\nName"`
+}
+
+type treeAnalyzer struct {
+	rtManager            artifactory.ArtifactoryServicesManager
+	extractPoliciesRegex *regexp.Regexp
+	rtAuth               auth.ServiceDetails
+	httpClientDetails    httputils.HttpClientDetails
+	url                  string
+	repo                 string
+	tech                 coreutils.Technology
+	parallelRequests     int
+}
+
+type CurationAuditCommand struct {
+	PackageManagerConfig *rtUtils.RepositoryConfig
+	extractPoliciesRegex *regexp.Regexp
+	workingDirs          []string
+	OriginPath           string
+	parallelRequests     int
+	*utils.GraphBasicParams
+}
+
+func NewCurationAuditCommand() *CurationAuditCommand {
+	return &CurationAuditCommand{
+		extractPoliciesRegex: regexp.MustCompile(extractPoliciesRegexTemplate),
+		GraphBasicParams:     &utils.GraphBasicParams{},
+	}
+}
+
+func (ca *CurationAuditCommand) setPackageManagerConfig(pkgMangerConfig *rtUtils.RepositoryConfig) *CurationAuditCommand {
+	ca.PackageManagerConfig = pkgMangerConfig
+	return ca
+}
+
+func (ca *CurationAuditCommand) SetWorkingDirs(dirs []string) *CurationAuditCommand {
+	ca.workingDirs = dirs
+	return ca
+}
+
+func (ca *CurationAuditCommand) SetParallelRequests(threads int) *CurationAuditCommand {
+	ca.parallelRequests = threads
+	return ca
+}
+
+func (ca *CurationAuditCommand) Run() (err error) {
+	rootDir, err := os.Getwd()
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	if len(ca.workingDirs) > 0 {
+		defer func() {
+			err = errorutils.CheckError(os.Chdir(rootDir))
```

Comment:
The above code overrides the error returned by the command. Should be changed to - 
```
			if e := errorutils.CheckError(os.Chdir(rootDir)); err == nil {
				err = e	
			}
```